src/comands.h

```diff
@@ -0,0 +1,93 @@
+#ifndef COMANDS_H_
+#define COMANDS_H_
+#include <json/json.h>
+#include <string>
+#include "types.h"
+
+namespace Comand {
+
+class BaseComand {
+ public:
+  std::string variant;
+  Json::Value toJson();
+};
+
+class Authenticate : public BaseComand {
+ public:
+  Authenticate(const data::ClientCredentials& client_credentials_in);
+  data::ClientCredentials client_credentials;
+  std::string toJson();
```

Comment:
It's better to avoid this. Though the different return type will be caught at compile...? Refactor later.